oxidation/client/src/components/SettingsModal.vue

```diff
@@ -0,0 +1,344 @@
+              <ion-item>
+                <ion-select
+                  interface="popover"
+                  :value="config.theme"
+                  :label="$t('SettingsPage.theme.label')"
+                  @ion-change="changeTheme($event.detail.value)"
+                >
+                  <ion-select-option value="dark">
+                    {{ $t('SettingsPage.theme.dark') }}
+                  </ion-select-option>
+                  <ion-select-option value="light">
+                    {{ $t('SettingsPage.theme.light') }}
+                  </ion-select-option>
+                  <ion-select-option value="system">
+                    {{ $t('SettingsPage.theme.system') }}
+                  </ion-select-option>
+                </ion-select>
+              </ion-item>
+            </ion-list>
+          </div>
+          <!-- advanced -->
+          <div
+            v-else
+            class="settings-advanced"
+          >
+            <ion-list class="settings-list">
+              <!-- send error report -->
+              <settings-option
+                :title="$t('SettingsPage.enableTelemetry')"
+                :description="$t('SettingsPage.enableTelemetryDescription')"
+                v-model="config.enableTelemetry"
+              />
+              <!-- minimise in status bar -->
+              <settings-option
+                :title="$t('SettingsPage.minimizeToSystemTray')"
+                :description="$t('SettingsPage.minimizeToSystemTrayDescription')"
+                v-model="config.minimizeToTray"
+              />
+              <!-- display unsync files -->
+              <settings-option
+                :title="$t('SettingsPage.unsyncFiles')"
+                :description="$t('SettingsPage.unsyncFilesDescription')"
+                v-model="config.unsyncFiles"
+              />
+            </ion-list>
+          </div>
+        </div>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script setup lang = "ts" >
+import {
+  IonPage,
+  IonHeader,
+  IonToolbar,
+  IonTitle,
+  IonButtons,
+  IonButton,
+  IonContent,
+  IonList,
+  IonRadioGroup,
+  IonItem,
+  IonRadio,
+  IonText,
+  IonIcon,
+  IonSelect,
+  IonSelectOption,
+  modalController
+} from '@ionic/vue';
+
+import {
+  close,
+  cog,
+  options
+} from 'ionicons/icons';
+import { ref, inject, toRaw, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { onMounted, onUnmounted } from '@vue/runtime-core';
+import { toggleDarkMode } from '@/states/darkMode';
+import { Config, StorageManager } from '@/services/storageManager';
+import { storageManagerKey } from '@/main';
+import SettingsOption from './SettingsOption.vue';
+
+const { locale } = useI18n();
+const storageManager = inject(storageManagerKey)!;
+const config = ref<Config>(structuredClone(StorageManager.DEFAULT_CONFIG));
+const showTOS = ref('general');
```

Comment:
Instead of using hard coded string, we could use an enum here. Something like

```typescript
enum SettingsPage {
  General = 'General',
  Advanced = 'Avanced'
}

const showTOS = ref(SettingsPage.General);

// Check what page we're on
if (showTOS == SettingsPage.General) {
  ...
}
```